### PR TITLE
Refactored PeepThoughtType to use strong enum.

### PR DESCRIPTION
**Pull request by Sciencentistguy**
Closes issue #12387

### CODE REVIEW

File: src/openrct2/peep/Peep.cpp [hunk 1]
```diff
@@ -1532,7 +1532,7 @@ static constexpr const uint8_t tshirt_colours[] = {
  */
 void Peep::InsertNewThought(PeepThoughtType thoughtType, uint8_t thoughtArguments)
 {
-    PeepActionType newAction = PeepThoughtToActionMap[thoughtType].action;
+    PeepActionType newAction = PeepThoughtToActionMap[static_cast<uint8_t>(thoughtType)].action;
```
Comment (duncanspumpkin) [inline src/openrct2/peep/Peep.cpp:1535]: We use `EnumValue` function to get the underlying value.

File: src/openrct2/peep/Peep.h [hunk 2]
```diff
@@ -74,148 +74,148 @@ enum class StaffType : uint8_t
...
-
-    PEEP_THOUGHT_TYPE_HELP = 168,                 // "Help! Put me down!"
-    PEEP_THOUGHT_TYPE_RUNNING_OUT = 169,          // "I'm running out of cash!"
-    PEEP_THOUGHT_TYPE_NEW_RIDE = 170,             // "Wow! A new ride being built!"
-    PEEP_THOUGHT_TYPE_NICE_RIDE_DEPRECATED = 171, // "Nice ride! But not as good as the Phoenix..."
-    PEEP_THOUGHT_TYPE_EXCITED_DEPRECATED = 172,   // "I'm so excited - It's an Intamin ride!"
-    PEEP_THOUGHT_TYPE_HERE_WE_ARE = 173,          // "...and here we are on X!"
-
-    PEEP_THOUGHT_TYPE_NONE = 255
+    CantAfford_0 = 0,     // "I can't afford"
```
Comment (duncanspumpkin) [inline src/openrct2/peep/Peep.h:79]: CantAffordRide

File: src/openrct2/peep/Peep.h [hunk 3]
```diff
@@ -74,148 +74,148 @@ enum class StaffType : uint8_t
...
+    VerySick = 3,         // "I feel very sick"
+    MoreThrilling = 4,    // "I want to go on something more thrilling than X"
+    Intense = 5,          // "X looks too intense for me"
+    HaventFinished = 6,   // "I haven't finished my X yet"
+    Sickening = 7,        // "Just looking at X makes me feel sick"
+    BadValue = 8,         // "I'm not paying that much to go on X"
+    GoHome = 9,           // "I want to go home"
+    GoodValue = 10,       // "X is really good value"
+    AlreadyGot = 11,      // "I've already got"
+    CantAfford = 12,      // "I can't afford"
```
Comment (duncanspumpkin) [inline src/openrct2/peep/Peep.h:91]: CantAffordItem

File: src/openrct2/peep/Peep.h [hunk 4]
```diff
@@ -74,148 +74,148 @@ enum class StaffType : uint8_t
...
+    FruitJuiceMuch = 148,
+    SoybeanMilkMuch = 149,
+    SuJongkwaMuch = 150,
+    SubSandwichMuch = 151,
+    CookieMuch = 152,
+
+    RoastSausageMuch = 156,
+
+    Help = 168,               // "Help! Put me down!"
+    RunninOut = 169,          // "I'm running out of cash!"
```
Comment (duncanspumpkin) [inline src/openrct2/peep/Peep.h:212]: Runnin**g**Out

File: src/openrct2/peep/Peep.h [hunk 5]
```diff
@@ -74,148 +74,148 @@ enum class StaffType : uint8_t
...
+    IcedTeaMuch = 141,
+    FunnelCakeMuch = 142,
+    SunglassesMuch = 143,
+    BeefNoodlesMuch = 144,
+    FriedRiceNoodlesMuch = 145,
+    WontonSoupMuch = 146,
+    MeatballSoupMuch = 147,
+    FruitJuiceMuch = 148,
+    SoybeanMilkMuch = 149,
+    SuJongkwaMuch = 150,
```
Comment (duncanspumpkin) [inline src/openrct2/peep/Peep.h:205]: SujongkwaMuch

File: src/openrct2/peep/Peep.h [hunk 6]
```diff
@@ -74,148 +74,148 @@ enum class StaffType : uint8_t
...
+    IcedTea = 109,
+    FunnelCake = 110,
+    Sunglasses = 111,
+    BeefNoodles = 112,
+    FriedRiceNoodles = 113,
+    WontonSoup = 114,
+    MeatballSoup = 115,
+    FruitJuice = 116,
+    SoybeanMilk = 117,
+    SuJongkwa = 118,
```
Comment (duncanspumpkin) [inline src/openrct2/peep/Peep.h:185]: Sujongkwa

**Comment by duncanspumpkin** [comment]
You seem to have a different version of clang format or something as you've changed a lot of unrelated files in this.

**Comment by Sciencentistguy** [comment]
Fixed the issues you mentioned. Not sure what happened to my clang-format there but I've removed all the changes to unrelated files.

**Review by duncanspumpkin — APPROVED** [review]

File: src/openrct2/peep/Peep.cpp [hunk 7]
```diff
@@ -1996,9 +1996,9 @@ bool Peep::SetName(const std::string_view& value)
  */
 void peep_thought_set_format_args(const rct_peep_thought* thought, Formatter& ft)
 {
-    ft.Add<rct_string_id>(PeepThoughts[thought->type]);
+    ft.Add<rct_string_id>(PeepThoughts[static_cast<uint8_t>(thought->type)]);
```
Comment (Gymnasiast) [inline src/openrct2/peep/Peep.cpp:1999]: EnumValue()

File: src/openrct2/peep/Guest.cpp [hunk 8]
```diff
@@ -1571,8 +1571,9 @@ bool Guest::DecideAndBuyItem(Ride* ride, ShopItem shopItem, money32 price)
                     // "I'm not paying that much for x"
                     PeepThoughtType thought_type = static_cast<PeepThoughtType>(
                         shopItem >= ShopItem::Photo2
-                            ? PEEP_THOUGHT_TYPE_PHOTO2_MUCH + static_cast<uint8_t>(shopItem - ShopItem::Photo2)
```
Comment (Gymnasiast) [inline src/openrct2/peep/Guest.cpp:1574]: This was already a mess, but this does not improve things.

I don't think the adding and subtracting should still be necessary, but @duncanspumpkin can probably tell for sure.
But even then, EnumValue() will be ever so slightly cleaner.

Comment (duncanspumpkin) [reply src/openrct2/peep/Guest.cpp:1574]: Its changed in my PR so don't worry about it.

Comment (Gymnasiast) [reply src/openrct2/peep/Guest.cpp:1574]: Hm, https://github.com/OpenRCT2/OpenRCT2/pull/13513 already makes great changes to this. Probably fine to leave this as-is and make sure it gets tackled there.

**Review by Gymnasiast — APPROVED** [review]
May be merged once the CIs are green.

**Pull request merged**